src/App/Security/Authorization/Voter/PrivateDiscussionVoter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Security\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Doctrine\Bundle\DoctrineBundle\Registry as Doctrine;
+
+class PrivateDiscussionVoter implements VoterInterface
+{
+    const VIEW       = 'view';
+    const EDIT       = 'edit';
+    const ADD_MEMBER = 'add_member';
+    const UNREADER   = 'unreader';
+
+    private $doctrine;
+
+    public function __construct(Doctrine $doctrine)
+    {
+        $this->doctrine = $doctrine;
+    }
+
+    public function supportsAttribute($attribute)
+    {
+        return in_array($attribute, array(
+            self::VIEW,
+            self::EDIT,
+            self::ADD_MEMBER,
+            self::UNREADER,
+        ));
+    }
+
+    public function supportsClass($class)
+    {
+        $supportedClass = 'App\Entity\PrivateDiscussion';
+
+        return $supportedClass === $class || is_subclass_of($class, $supportedClass);
+    }
+
+    public function vote(TokenInterface $token, $discussion, array $attributes)
+    {
+        // check if class of this object is supported by this voter
+        if (!$this->supportsClass(get_class($discussion))) {
+            return VoterInterface::ACCESS_ABSTAIN;
+        }
+
+        // check if the voter is used correctly, only allow one attribute
+        if (1 !== count($attributes)) {
+            throw new \InvalidArgumentException(
+                'Only one attribute is allowed'
+            );
+        }
+
+        // set the attribute to check against
+        $attribute = $attributes[0];
+
+        // check if the given attribute is covered by this voter
```

Comment:
Same as before. Useless test. The voter won't be called if the attribute is not supported.